### PR TITLE
Added Pace Curve support for running, swimming and cycling.

### PR DESCRIPTION
**Pull request by brendankntb**
Added a class to create Pace Curves similar to the existing Power Curve functionality. Extended it to support running, swimming and cycling. I also cerated a unit test class to ensure that it is working for all conditions currently supported.

### CODE REVIEW

File: Cycling Analysis/pace_curve.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import pandas as pd
```
Comment (pranav5722222) [inline Cycling Analysis/pace_curve.py:1]: Import statements are fine and don't cause any concerns

File: Cycling Analysis/pace_curve.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.interpolate import make_interp_spline
+
+class PaceCurveAnalyzer:
```
Comment (pranav5722222) [inline Cycling Analysis/pace_curve.py:6]: The class is clearly defined with clear documentation.

File: Cycling Analysis/pace_curve.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        format_duration: Formats the duration in seconds to a human-readable format.
+        create_pace_curve: Creates a pace curve based on activity type, date, and number of days.
+        plot_pace_curve: Plots the pace curve.
+
+    """
+
+    def __init__(self, data_source=None):
+        self.data_source = data_source
+        if self.data_source is None:
+            self.data_source = pd.read_csv('data/extended_activities_TRI001.csv')
```
Comment (pranav5722222) [inline Cycling Analysis/pace_curve.py:26]: Using absolute file paths is a good method of avoiding any problems.

File: Cycling Analysis/pace_curve.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        numeric_durations = np.arange(len(filtered_durations))
+
+        # Convert the list to a Pandas Series for easy rolling mean computation
+        pace_series = pd.Series(filtered_paces)
+
+        # Compute the moving average (rolling mean)
+        # The window size determines the smoothing level
+        smooth_paces = pace_series.rolling(window=2, center=True).mean().fillna(pace_series)
+
+        # Plotting the original and smoothed curve
```
Comment (pranav5722222) [inline Cycling Analysis/pace_curve.py:137]: The pace curve is plotted by the method, and as it mainly deals with visualisation, the code appears perfect.

**Review by MorgaineBarter — APPROVED** [review]

**Pull request merged**